crypto/bn/bn_mont.c

```diff
@@ -400,6 +404,13 @@ int BN_MONT_CTX_set(BN_MONT_CTX *mont, const BIGNUM *mod, BN_CTX *ctx)
     mont->RR.top = ret;
     mont->RR.flags |= BN_FLG_FIXED_TOP;
 
+    BN_mod_mul_montgomery(&mont->RRR, &mont->RR, &mont->RR, mont, ctx);
+
+    BN_mod_mul_montgomery(&mont->RR16, &mont->RRR, &mont->RRR, mont, ctx);
+    for (j = 0; j < (15 - 1); j++) {
+    	BN_mod_mul_montgomery(&mont->RR16, &mont->RR16, &mont->RR16, mont, ctx);
+    }
```

Comment:
This slows down the construction of every BN_MONT_CTX even for cases where RSA public key operations aren't being done.

Also, it is very common that an RSA public key is constructed, used to verify a single signature, and then thrown away. E.g. many X.509 verification operations are like this. Won't this extra work actually slow down such uses?

Reply:
Agreed! Removed.